docs/src/main/asciidoc/security-architecture-concept.adoc

```diff
@@ -10,11 +10,28 @@ include::_attributes.adoc[]
 
 The Quarkus Security architecture provides several built-in authentication mechanisms. The `HttpAuthenticationMechanism` interface is the main entry mechanism for securing HTTP applications in Quarkus. Quarkus Security is also highly customizable.
 
-== Core components of Quarkus Security
+== Overview of the Quarkus Security architecture
 
+When a client sends an HTTP request, Quarkus Security orchestrates security authentication and authorization by interacting with several built-in core components including `HttpAuthenticationMechanism`, `IdentityProvider`, and `SecurityIdentityAugmentor`.
+
+The sequential security validation process results in one of three outcomes:
+
+* The HTTP request is authenticated and authorized and access to the Quarkus application is granted
+* The HTTP request authentication fails and the requester receives a challenge
```

Comment:
I couldn't find explanation of `challenge` in this guide or in any other. IMO sentence _requester receives a challenge_ is only going to confuse incomers. `security-customization.adoc` is dealing challenge, maybe you can point there for practical example of what you mean?